packages/thirdweb/src/chainsaw/endpoints/getTransactions.ts

```diff
@@ -0,0 +1,109 @@
+import type { ThirdwebClient } from "../../client/client.js";
+import type { Address } from "../../utils/address.js";
+import { getClientFetch } from "../../utils/fetch.js";
+import type { Prettify } from "../../utils/type-utils.js";
+import { formatChainsawTransactions } from "../formatter.js";
+import { addRequestPagination } from "../paging.js";
+import type {
+  ChainsawPagingParams,
+  ChainsawResponse,
+  ChainsawTransactions,
+  Transactions,
+} from "../types.ts";
+import { getTransactionsEndpoint } from "../urls.js";
+
+export type GetTransactionsParams = Prettify<
+  {
+    /**
+     * A client is the entry point to the thirdweb SDK. It is required for all other actions.
+     *
+     * You can create a client using the `createThirdwebClient` function.
+     * Refer to the [Creating a Client](https://portal.thirdweb.com/typescript/v5/client) documentation for more information.
+     *
+     */
+    client: ThirdwebClient;
+    /**
+     * Contract address to fetch transactions for
+     */
+    to: Address;
+    /**
+     * Chain IDs to fetch transactions from. If omitted, will look on all supported chains
+     */
+    chainIds?: number[];
+    /**
+     * Start of the date range to search in
+     */
+    startDate?: Date;
+    /**
+     * End of the date range to search in
+     */
+    endDate?: Date;
+  } & ChainsawPagingParams
+>;
+
+export type GetTransactionsResult = {
+  transactions: ChainsawTransactions;
+  page?: number;
+};
+
+/**
+ * @beta
+ *
+ * Get transactions to a contract
+ *
+ * @param {GetTransactionsParams} params
+ * @returns {Promise<GetTransactionsResult>}
+ *
+ * @example
+ * ```ts
+ * import { createThirdwebClient } from "thirdweb";
+ * import { getTransactions } from "thirdweb/chainsaw";
+ *
+ * const client = createThirdwebClient({ clientId: "..." });
+ * const startDate = new Date(Date.now() - 24 * 60 * 60_000);
```

Comment:
You don't have to but I'd recommend adding another example without the dates. If we show them in the example everyone will assume they're required all the time

Reply:
Done in [9121631](https://github.com/thirdweb-dev/js/pull/4004/commits/9121631ff9ec180b3f234a9ef113f141803a5bad)